profile/src/main/java/com/scottlogic/deg/profile/dto/ProfileSchemaFileLoader.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.profile.dto;
+
+import com.scottlogic.deg.common.ValidationException;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Used to validate a DataHelix Profile JSON file.
+ * <p>
+ * Checks that the profile JSON file is valid against the DataHelix Profile Schema (datahelix.schema.json)
+ */
+public class ProfileSchemaFileLoader implements ProfileSchemaLoader {
+    private final ProfileSchemaValidator validator;
+
+    public ProfileSchemaFileLoader(ProfileSchemaValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public void validateProfile(File profileFile, URL schemaUrl) {
+        String schema;
+        if (schemaUrl == null) {
+            throw new ValidationException("Null Schema");
+        }
+        try {
+            byte[] data = Files.readAllBytes(profileFile.toPath());
```

Comment:
I'd like the file read to string method to be separated into its own small method, so the validateProfile method only looks like
```
if url = null
   throw exception
string schema = readSchema
try
   validateSchema
catch
   throw validation exception
```

to keep the top level method abstract, and hide the details behind less and less abstract methods